webpack.config.js

```diff
@@ -107,10 +133,22 @@ const config = {
 			{
 				test: /\.jsx?$/,
 				enforce: 'post',
-				use: {
-					loader: 'babel-loader',
-					options: babelSettings,
-				},
+				use: [
+					{
+						loader: 'thread-loader',
+						options: {
+							workers: Math.max( 2, Math.floor( os.cpus().length / 2 ) ),
+						},
+					},
+					{
+						loader: 'babel-loader',
+						options: {
+							configFile: path.resolve( __dirname, 'node_modules', 'wp-calypso', 'babel.config.js' ),
+							cacheDirectory: true,
+							cacheIdentifier: require( 'wp-calypso/server/bundler/babel/babel-loader-cache-identifier' ),
```

Comment:
Do we need our own cache identifier? If not, should this be "client" instead of "server"?

Reply:
This is the identifier that Calypso uses on its **client** config, see: [[#](https://github.com/Automattic/wp-calypso/blob/master/webpack.config.js#L167)] [[#](https://github.com/Automattic/wp-calypso/blob/master/webpack.config.js#L26)]. The identifier itself is quite simple, it's like the default one but it also takes into account the version of a custom babel transform that Calypso uses: [see here](https://github.com/Automattic/wp-calypso/blob/cf1d19c8e494c110eb400d0c8fdf43a94c81f9b3/server/bundler/babel/babel-loader-cache-identifier/index.js).

It's confusing that it's under `server/bundler`, but it's really used by the client. It may be overkill for us, but I prefer to apply the "do whatever Calypso does and forget about it" rule of thumb.

Reply:
Ah. yeah - I totally missed that this was in a `require()`. Nevermind!